### PR TITLE
chore: make CI config DRY

### PR DESCRIPTION
**Pull request by bartlomieju**
[Reference](https://github.com/denoland/deno_std/pull/453?fbclid=IwAR1U5OVwaXc1a4vpW6k7Nerv501D5NocrffZoIf4WdGapkNOKVMoQ090mUk#discussion_r288976144)

Using templates to configure CI to dry up configuration

### CODE REVIEW

File: .ci/steps.unix.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+steps:
+  - script: npm install eslint typescript@$(TS_VERSION) @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint-config-prettier
+  - script: curl -L https://deno.land/x/install/install.sh | sh -s $(DENO_VERSION)
+  - script: echo '##vso[task.prependpath]$(HOME)/.deno/bin/'
+  - script: npx eslint **/*.ts --max-warnings=0
+  - script: ${{ parameters.exe_name }} run --allow-run --allow-write --allow-read format.ts --check
+  - script: ${{ parameters.exe_name }} run --allow-run --allow-net --allow-write --allow-read --config=tsconfig.test.json test.ts
```
Comment (bartlomieju) [inline .ci/steps.unix.yml]: First and 3 last commands should be templated as well

File: .ci/steps.win.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+steps:
```
Comment (ry) [inline .ci/steps.win.yml]: I'd refer `ci` directory to `.ci`

Comment (bartlomieju) [reply .ci/steps.win.yml]: I thought about it, but won't it be confusing for users? It'll look like one of the available modules.

Comment (ry) [reply .ci/steps.win.yml]: It's still listed in github, which is where 99% of people will ever see this repo's source.

Comment (ry) [reply .ci/steps.win.yml]: ok good point

**Comment by bartlomieju** [comment]
@ry this should be ready for review

File: azure-pipelines.yml [hunk 3]
```diff
@@ -2,39 +2,24 @@ variables:
...
     steps:
-      - bash: npm install eslint typescript@$(TS_VERSION) @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint-config-prettier
-      - powershell: iwr https://deno.land/x/install/install.ps1 -out install.ps1; .\install.ps1 $(DENO_VERSION)
-      - bash: echo "##vso[task.prependpath]C:\Users\VssAdministrator\.deno\\bin"
-      - bash: npx eslint **/*.ts --max-warnings=0
-      - bash: deno.exe run --allow-run --allow-write --allow-read format.ts --check
-      - bash: deno.exe run --allow-run --allow-net --allow-write --allow-read --config=tsconfig.test.json test.ts
+      - template: .ci/template.windows.yml
+        parameters:
+          exe_suffix: ".exe"
```
Comment (ry) [inline azure-pipelines.yml]: Maybe the parameters should be moved to last line of template.windows.yml ?

File: .ci/template.windows.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+steps:
+  - bash: npm install eslint typescript@$(TS_VERSION) @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint-config-prettier
+  - powershell: iwr https://deno.land/x/install/install.ps1 -out install.ps1; .\install.ps1 $(DENO_VERSION)
+  - bash: echo "##vso[task.prependpath]C:\Users\VssAdministrator\.deno\\bin"
+  - bash: npx eslint **/*.ts --max-warnings=0
```
Comment (ry) [inline .ci/template.windows.yml]: Can the eslint be moved to common?

Comment (bartlomieju) [reply .ci/template.windows.yml]: Nope, it's run with `npx` on Linux and windows, doesn't work on Mac :(

**Pull request merged**